src-tool/Pact/Analyze/Eval/Core.hs

```diff
@@ -407,6 +412,21 @@ evalCore (ObjTake argTy _keys obj) = withSing argTy $ do
 evalCore (ObjLength (SObjectUnsafe (SingList hlist)) _obj) = pure $ literalS $
   hListLength hlist
 
+evalCore (IsPrincipal p) = do
+  p' <- eval p
+  case unliteralS p' of
+    Nothing -> let (S _ str) = coerceS @Str @String p'
+               in pure $ sansProv (literal "internal-principal-" `SBVS.isPrefixOf` str)
```

Comment:
Why is it `internal-principle-*` rather than `is-principal` here?

Reply:
I intended to emphasize that the string "internal-principal-`?`" is utilized exclusively within the system. It is created by the function "create-principal" which takes the relevant cap id `?` as input.

Reply:
Perhaps just a comment to this effect, then, because it doesn't seem to match the pattern of the other definitions.